pkg/plugins/log/logger.go

```diff
@@ -42,3 +44,13 @@ func (d *grafanaInfraLogWrapper) Warn(msg string, ctx ...any) {
 func (d *grafanaInfraLogWrapper) Error(msg string, ctx ...any) {
 	d.l.Error(msg, ctx...)
 }
+
+func (d *grafanaInfraLogWrapper) FromContext(ctx context.Context) Logger {
+	concreteInfraLogger, ok := d.l.FromContext(ctx).(*log.ConcreteLogger)
```

Comment:
I agree it looks weird, but I couldn't find a way around it.

https://github.com/grafana/grafana/blob/7e01dde7621808dfb7a1cc1c30ca631d163a3ad4/pkg/infra/log/log.go#L203

`FromContext` returns a pkg/infra `Logger` interface, but if I replace this method with:

```go
func (d *grafanaInfraLogWrapper) FromContext(ctx context.Context) Logger {
	return d.l.FromContext(ctx)
}
``` 

it doesn't compile:

```
Cannot use 'd.l.FromContext(ctx)' (type "github.com/grafana/grafana/pkg/infra/log".Logger) as the type "github.com/grafana/grafana/pkg/plugins/log".Logger Type does not implement 'Logger' as some methods are missing: New(ctx ...any) Logger
```

This is because the pkg/infra `Logger` has this signature for `New`:

https://github.com/grafana/grafana/blob/025b2f3011821bb7e6ab4781e252f3a1ecd19130/pkg/infra/log/interface.go#L17

But the plugins logger returns a `Logger` interface instead:

https://github.com/grafana/grafana/blob/605af9af8e0671fb7053361de2017b4dd1dd610d/pkg/plugins/log/ifaces.go#L8

So the two interfaces are not compatible

I tried changing the pkg/infra interface so it returns a `Logger` interface rather than a `*ConcreteLogger`, but I got type errors all over the place so I think we can eventually do it as a follow-up work

Reply:
Hmm okay